EIPS/eip-4834.md

```diff
@@ -0,0 +1,40 @@
+---
+eip: 4834
+title: Hierarchical Domains Standard
+description: The Heirarchal Domains Standard is a standard for extremely generic name resolution.
+author: Pandapip1 (@Pandapip1)
+discussions-to: https://ethereum-magicians.org/t/erc-4834-hierarchical-domains-standard/8388
+status: Draft
+type: Standards Track
+category: ERC
+created: 2022-02-22
+requires: 165
+---
+
+## Abstract
+The Hierarchical Domains Standard is a standard for extremely generic name resolution. conversely it makes it simple to extend - any smart contract that implements the ERC-4834 interface could be set as a subdomain. This differs from ERC-137 because it does not provide a separate "resolver" contract - any behaviour should be part of the ERC-4834 contract itself.
+
+## Motivation
+The motivation behind this ERC is that existing standards like ERC-137 are relatively complex and have limitations. The advantage of ERC-4834 is that it provides a minimal interface that supports name resolution. This permits use-cases such as referencing an ERC-20 or ERC-1155 token by a name directly, without using a "resolver."
+
+This ERC also implements flexible access control. This allows for clever implementations, such as having a token that can be redeemed for subdomains of the token, and domains that are themselves NFTs that grant the ability to manage the domain to the owner of said NFT. There can also be multiple managers of a single domain with different access, instead of a single "owner."
+
+In addition, TLDs (like .com), subdomains, and even the "root" itself (analogous to the ENS registry) are all implemented as ERC-4834 smart contracts, meaning that name-resolution is a relatively simple recursive algorithm, and there is no limit to the amount of nesting that is possible (e.g. 0.1.2.3.4.5.6.7.8.9.a.b.c.d.e.f.g.h.i.j.k.l.m.n.o.p.q.r.s.t.u.v.w.x.y.z could be valid if the root contains z, and z contains y, and etc).
+
+## Specification
+Solidity Interface (with descriptions): [eip-4834/IDomain.sol]
```

Comment:
The specification should be inline.  You should be able to do the following in order to get syntax highlighting.
````
```sol
<specification here>
```
````

Reply:
Inlined it

Reply:
I'm going to unresolve this so I remember to put the inline interface back.